File: pm_core/src/infrastructure/store/model/priority_scheme_item_model.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package model

type PrioritySchemeItemModel struct {
	BaseModel
	TenantID   int64 `gorm:"column:tenant_id;not null" json:"tenantId"`
	SchemeID   int64 `gorm:"column:scheme_id;not null" json:"schemeId"`
	PriorityID int64 `gorm:"column:priority_id;not null" json:"priorityId"`
	Sequence   int   `gorm:"column:sequence;not null" json:"sequence"`
}

func (PrioritySchemeItemModel) TableName() string {
	return "priority_scheme_items"
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/model/PrioritySchemeItemModel.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.SQLRestriction;

@Entity
@Table(name = "priority_scheme_items")
@SQLRestriction("deleted_at IS NULL")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@SuperBuilder
public class PrioritySchemeItemModel extends BaseModel {

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(name = "scheme_id", nullable = false)
    private Long schemeId;

    @Column(name = "priority_id", nullable = false)
    private Long priorityId;

    @Column(name = "sequence")
    private Integer sequence;
}
